ckan/views/dataset.py

```diff
@@ -510,137 +477,63 @@ def read(package_type: str, id: str) -> Union[Response, str]:
-                    url = h.url_for(
-                        u'{}_resource.new'.format(package_type),
-                        id=pkg_dict[u'name']
-                    )
+            if create_on_ui_requires_resources:
+                data_dict['state'] = 'draft'
+            # required for setting state = draft in ignore_not_package_admin
+            context['allow_state_change'] = True
+
+            data_dict['type'] = package_type
+            pkg_dict = get_action('package_create')(context, data_dict)
+
+            for plugin in plugins.PluginImplementations(plugins.IFormRedirect):
+                url = plugin.dataset_save_redirect(
+                    package_type, pkg_dict['name'], 'create', save_action,
+                    data_dict)
+                if url:
                     return h.redirect_to(url)
 
-                get_action(u'package_update')(
-                    Context(context, allow_state_change=True),
-                    dict(pkg_dict, state=u'active')
-                )
-                return h.redirect_to(
-                    u'{}.read'.format(package_type),
-                    id=pkg_dict["id"]
+            if create_on_ui_requires_resources:
+                url = h.url_for(
+                    '{}_resource.new'.format(package_type),
+                    id=pkg_dict['name']
                 )
+                return h.redirect_to(url)
 
-            return _form_save_redirect(
-                pkg_dict[u'name'], u'new', package_type=package_type
+            return h.redirect_to(
+                '{}.read'.format(package_type),
+                id=pkg_dict["id"]
             )
         except NotAuthorized:
-            return base.abort(403, _(u'Unauthorized to read package'))
+            return base.abort(403, _('Unauthorized to read package'))
         except NotFound:
-            return base.abort(404, _(u'Dataset not found'))
-        except SearchIndexError as e:
-            try:
-                exc_str = str(repr(e.args))
-            except Exception:  # We don't like bare excepts
-                exc_str = str(str(e))
-            return base.abort(
-                500,
-                _(u'Unable to add package to search index.') + exc_str
-            )
+            return base.abort(404, _('Dataset not found'))
         except ValidationError as e:
             errors = e.error_dict
             error_summary = e.error_summary
-            if is_an_update:
-                # we need to get the state of the dataset to show the stage we
-                # are on.
-                pkg_dict = get_action(u'package_show')(context, data_dict)
-                data_dict[u'state'] = pkg_dict[u'state']
-                return EditView().get(
-                    package_type,
-                    data_dict[u'id'],
-                    data_dict,
-                    errors,
-                    error_summary
-                )
-            data_dict[u'state'] = u'none'
```

Comment:
Is this old unused code?

Reply:
Right now the bulk of the CreateView and EditView code duplicate the same editing functionality. Here I've removed the editing functionality from CreateView so we can rely on the EditView implementation instead of having two slightly different copies of the same code.